chore: add codebyte examples on arrays.md entry

tsirianni opened the pull request:
### Description

Adds Codebyte blocks to offer executable examples on how to create and print an array to the console and how to access elements from an array. Also makes some changes on the descriptions and added new ways of creating an array.

Closes #4136 

### Type of Change

- Editing an existing entry (fixing a typo, bug, issues, etc)
- Updating the documentation

### Checklist

- [X] All writings are my own.
- [X] My entry follows the Codecademy Docs style guide.
- [X] My changes generate no new warnings.
- [X] I have performed a self-review of my own writing and code.
- [X] I have checked my entry and corrected any misspellings.
- [X] I have made corresponding changes to the documentation if needed.
- [X] I have confirmed my changes are not being pushed from my forked `main` branch.
- [X] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [X] Under "Development" on the right, I have linked any issues that are relevant to this PR

### Additional Information

- Modified introduction to bold the first occurrence of the term **Array** as per style guide and template suggestions
- Modified syntax to only provide an idea of how to create an array and leave the other possible ways (e.g. Class) to the **Creating an array** section
- Added new examples of array creation (receiving variables instead of primitive values as elements and empty array)
- Codebyte blocks had their language modified from `codebyte/js` to `codebyte/javascript` since it is the suggested form on `style-guide.md`
- I tried to keep all the Codebyte examples around the same theme `music genres`, as an attempt to facilitate the understanding of the new capabilities being presented to the learners. Not sure if this makes sense, but it creates a nice consistency in the examples!

If I have in any way overstepped and made more modification than necessary or if any modifications are not acceptable/do not make sense, please let me know and I will gladly revert these changes!

Also, I have not found in the suggested documents whether or not semantic commit messages are used so I ended up creating this one as a `chore` but if needed I can also modify the commit message!

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Codecademy/docs?pullRequest=4275) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Codecademy/docs?pullRequest=4275) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Codecademy/docs?pullRequest=4275) it.</sub>

ishg-153 commented on content/javascript/concepts/arrays/arrays.md:
```diff
@@ -98,40 +110,39 @@ Any object can be an element of an array, including other arrays. Arrays with on
-console.log(nestedArray[1][0][2]);
-// Output: C
+console.log(nestedArray[1][0][1]);
+// Output: Classic
```
```suggestion
```

ishg-153 commented on content/javascript/concepts/arrays/arrays.md:
```diff
@@ -98,40 +110,39 @@ Any object can be an element of an array, including other arrays. Arrays with on
 
 console.log(nestedArray[0][1][1]);
-// Output: 4
+// Output: Hip Hop
```
```suggestion
```

ishg-153 reviewed (changes requested):
@tsirianni I have requested a few changes, please make those. Also, make the following changes:
- Before the individual codebytes add a sentence that prompts the user to run the codeblock.

tsirianni commented:
@ishg-153, I have made the requested modifications and hopefully they are now aligned with your expectations. Please let me know if you would like any additional changes.

ishg-153 commented:
@tsirianni you need to address all of my suggestions.

**Before the individual codebytes add a sentence that prompts the user to run the codeblock.**

tsirianni commented:
@ishg-153 I am fairly certain that I have addressed all of your suggestions and that I have included a prompt sentence before each of the four executable codebyte blocks. Perhaps I am not seeing it, could you point it out which executable codebyte block does not have a prompt sentence before it?

ishg-153 commented on content/javascript/concepts/arrays/arrays.md:
````diff
@@ -26,29 +26,45 @@ const numberArray = [0, 1, 2, 3];
 
-```js
-const classArray = new Array(0, 1, 2, 3);
+The most straightforward way to create an array is using an array literal. Run the code block below to see how to create an array using an array literal and print it to the console:
````
```suggestion
Run the code block below to see how to create an array using an array literal and print it to the console:
```

ishg-153 commented on content/javascript/concepts/arrays/arrays.md:
````diff
@@ -26,29 +26,45 @@ const numberArray = [0, 1, 2, 3];
 ```
 
-## Creating an Array
+Whitespace is ignored and declarations can span multiple lines.
````
```suggestion
>**Note:**Whitespace is ignored and declarations can span multiple lines.
```

ishg-153 reviewed (changes requested):
@tsirianni I've added some direct comments you can make those changes.

tsirianni commented:
@ishg-153 I have made the suggested changes!

ishg-153 reviewed (approved):
@tsirianni Looks good for a second review.

Sriparno08 reviewed (approved):
Looks good 👍
